src/cares_wrap.cc

```diff
@@ -270,31 +283,43 @@ void ares_sockstate_cb(void* data,
 }
 
 
-Local<Array> HostentToAddresses(Environment* env, struct hostent* host) {
+Local<Array> HostentToAddresses(Environment* env,
+                                struct hostent* host,
+                                Local<Array> append_to = Local<Array>()) {
   EscapableHandleScope scope(env->isolate());
-  Local<Array> addresses = Array::New(env->isolate());
+  auto context = env->context();
+  Context::Scope context_scope(context);
+  bool append = !append_to.IsEmpty();
+  Local<Array> addresses = append ? append_to : Array::New(env->isolate());
+  size_t offset = addresses->Length();
 
   char ip[INET6_ADDRSTRLEN];
   for (uint32_t i = 0; host->h_addr_list[i] != nullptr; ++i) {
     uv_inet_ntop(host->h_addrtype, host->h_addr_list[i], ip, sizeof(ip));
     Local<String> address = OneByteString(env->isolate(), ip);
-    addresses->Set(i, address);
+    addresses->Set(context, i + offset, address).FromJust();
   }
 
-  return scope.Escape(addresses);
+  return append ? addresses : scope.Escape(addresses);
 }
 
 
-Local<Array> HostentToNames(Environment* env, struct hostent* host) {
+Local<Array> HostentToNames(Environment* env,
+                            struct hostent* host,
+                            Local<Array> append_to = Local<Array>()) {
   EscapableHandleScope scope(env->isolate());
-  Local<Array> names = Array::New(env->isolate());
+  auto context = env->context();
+  Context::Scope context_scope(context);
```

Comment:
I haven't looked in close detail at this PR but a `Context::Scope` is appropriate in libuv and c-ares callbacks.  You normally create them as early as possible though, in the callback itself.